dlc/src/main/scala/org/bitcoins/dlc/sign/DLCTxSigner.scala

```diff
@@ -71,299 +72,125 @@ case class DLCTxSigner(
-                if (witness == EmptyScriptWitness) {
-                  Future.failed(
-                    new RuntimeException(
-                      s"Funding Inputs must be SegWit: $utxo"))
-                } else {
-                  Future.successful(witness)
-                }
-              case _: NonWitnessTransaction =>
-                Future.failed(
-                  new RuntimeException(s"Funding Inputs must be SegWit: $utxo"))
-            }
-          }
-
-          sigFs += witnessF.flatMap {
-            case witness: ScriptWitnessV0 =>
-              Future.successful((utxo.outPoint, witness))
-            case witness: ScriptWitness =>
-              Future.failed(
-                new RuntimeException(s"Unrecognized script witness: $witness"))
-          }
-        }
-      }
-
-      sigs <- Future.sequence(sigFs.result())
-    } yield {
-      val sigsMap = sigs.toMap
-
-      val fundingInputs = if (isInitiator) {
-        offer.fundingInputs
-      } else {
-        accept.fundingInputs
-      }
-
-      val sigsVec = fundingInputs.map { input =>
-        input.outPoint -> sigsMap(input.outPoint)
-      }
-
-      FundingSignatures(sigsVec)
-    }
+  def signFundingTx(): Future[FundingSignatures] = {
+    DLCTxSigner.signFundingTx(builder.buildFundingTx, fundingUtxos)
   }
 
   /** Constructs the signed DLC funding transaction given remote FundingSignatures */
-  def signFundingTx(remoteSigs: FundingSignatures): Future[Transaction] = {
-    val fundingInputs = offer.fundingInputs ++ accept.fundingInputs
-
-    val psbtF = for {
-      localSigs <- createFundingTxSigs()
-      allSigs = localSigs.merge(remoteSigs)
-      fundingTx <- builder.buildFundingTx
-    } yield {
-      fundingInputs.zipWithIndex.foldLeft(
-        PSBT.fromUnsignedTxWithP2SHScript(fundingTx)) {
-        case (psbt, (fundingInput, index)) =>
-          val witness = allSigs(fundingInput.outPoint)
-
-          psbt
-            .addUTXOToInput(fundingInput.prevTx, index)
-            .addFinalizedScriptWitnessToInput(fundingInput.scriptSignature,
-                                              witness,
-                                              index)
-      }
-    }
-
-    psbtF.flatMap { psbt =>
-      val finalizedT = if (psbt.isFinalized) {
-        Success(psbt)
-      } else {
-        psbt.finalizePSBT
-      }
-
-      val txT = finalizedT.flatMap(_.extractTransactionAndValidate)
-      Future.fromTry(txT)
-    }
-  }
-
-  private def findSigInPSBT(
-      psbt: PSBT,
-      pubKey: ECPublicKey): Future[PartialSignature] = {
-    val sigOpt = psbt.inputMaps.head.partialSignatures
-      .find(_.pubKey == pubKey)
-
-    sigOpt match {
-      case None =>
-        Future.failed(new RuntimeException("No signature found after signing"))
-      case Some(partialSig) => Future.successful(partialSig)
-    }
+  def completeFundingTx(remoteSigs: FundingSignatures): Future[Transaction] = {
```

Comment:
Would be nice to have a 

```scala
def completeFundingTx(localSigs: FundingSignatures, remoteSigs: FundingSignatures): Transaction
```

Reply:
same for cet

Reply:
I'm guessing this is needed because of current `Sign` API, nadav is working on making a synchronous version in #2652